readthedocs/search/parsers.py

```diff
@@ -123,6 +123,59 @@ def _parse_sections(self, title, body):
         We can have pages that have content before the first title or that don't have a title,
         we index that content first under the title of the original page.
         """
+
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        seen_dls = []
+        for dl in dls:
+            if dl in seen_dls:
+                continue
+            for inner_dl in dl.css("dl"):
+                seen_dls.append(inner_dl)
```

Comment:
Is there a reason why we are adding the nested dls here? We want to index those too separately

Reply:
Thanks! This was from the former approach and I developed a blind spot for it... removing :+1:

Reply:
Splitting the whole thing into a method and yielding indexed nodes seems to work well :+1: